cmd/internal/ecs/ecs.go

```diff
@@ -127,9 +150,107 @@ func addAgentToECS(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	return nil
+}
+
+func addAgentToECS(cmd *cobra.Command, args []string) error {
+	err := checkAPIKeyAndProjectID()
+	if err != nil {
+		return err
+	}
+
 	return RunAddWorkflow()
 }
 
 func removeAgentFromECS(cmd *cobra.Command, args []string) error {
 	return fmt.Errorf("this command is not yet implemented")
 }
+
+func printCloudFormationFragment(cmd *cobra.Command, args []string) error {
+	err := checkAPIKeyAndProjectID()
+	if err != nil {
+		return err
+	}
+
+	const isEssential = false
+	agentContainer := makeAgentContainerDefinition(
+		optionals.None[string](),
+		optionals.None[string](),
+		optionals.None[string](),
+		isEssential,
+	)
+
+	result, err := json.MarshalIndent(agentContainer, "", "  ")
+	if err != nil {
+		return errors.Wrapf(err, "unable to format container definition as JSON")
+	}
+
+	fmt.Println(string(result))
+	return nil
+}
+
+func printECSTaskDefinition(cmd *cobra.Command, args []string) error {
+	err := checkAPIKeyAndProjectID()
+	if err != nil {
+		return err
+	}
+
+	const isEssential = true
+	agentContainer := makeAgentContainerDefinition(
+		optionals.None[string](),
+		optionals.None[string](),
+		optionals.None[string](),
+		isEssential,
+	)
+
+	// XXX Whereas the AWS SDK defines a `types.TaskDefinition`, that type is
+	// intended for use in calling the AWS APIs, not for printing JSON fragments
+	// for pasting into the AWS console. We therefore take the brittle approach of
+	// manually cobbling together an object that will serialize to the appropriate
+	// JSON value.
```

Comment:
Wow, yuck. The Task definition SDK's JSON does not match the JSON format accepted in the console?  I wondered whether there was some better option out there; Terraform, naturally, just uses the SDK, and the CDK uses CloudFormation format.

I think it would be better to define appropriate structs here (if we can't get them elsewhere), but it is non-blocking.

Reply:
Commit ec735e8 defines types in a separate package for both the AWS console and CloudFront.